MantidPlot/src/ProjectRecovery.cpp

```diff
@@ -661,6 +665,21 @@ void ProjectRecovery::saveWsHistories(const Poco::Path &historyDestFolder) {
   }
 }
 
+void ProjectRecovery::removeEmptyGroupsFromADS(
+    std::vector<boost::shared_ptr<Mantid::API::Workspace>> &wsHandles) {
+  for (auto i = 0u; i < wsHandles.size(); ++i) {
```

Comment:
Would be done better as a range based for loop.

Reply:
First iteration privately was, I changed from that because I needed an index to act upon on line 678 for wsHandles.erase() so I could give an accurate iterator.